### DISCUSSION TITLE
Review from Eli Miller

### DISCUSSION THREAD

**Discussion by Yousuf28** [phuse-org/sendigR#9] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Yousuf28** [comment]
- "You are calling @import and @importFrom in several places. Best practice is to [call all of these at once](https://github.com/tidyverse/dplyr/blob/main/R/dplyr.r) in the package documentation file."


01. ImportFrom used in initSENDFunctions.R file and NAMESPACE file. NAMESPACE file edited by roxygen package which added importFrom from initSENDFunction.R file. So basically importFrom used in only one R file.
<img width="1262" alt="Screen Shot 2022-07-01 at 4 23 20 PM" src="https://user-images.githubusercontent.com/31291716/177204947-9f877564-3e71-474f-b96f-ea2f2059bb2a.png">

**Comment by Yousuf28** [comment]
- "You should add BioCelerate to the DESCRIPTION file as the copyright holder. [This is required](https://github.com/atorus-research/xportr/blob/fdbd8d892b60ee18853483fdf13e1f910843cbdc/DESCRIPTION#L23) if you're submitting to CRAN."

2. BioCelerate added in Description file
[link](https://github.com/phuse-org/sendigR/commit/767d1192ba8e4671f7375eb8b491d9429bf6cd3f)

**Comment by Yousuf28** [comment]
- "In several places you're using print to send messages to the console. I would recommend to use message or warning to allow people to suppress messages if they need to."

3. print function- Here is the file list where print used.
<img width="642" alt="Screen Shot 2022-07-01 at 4 37 38 PM" src="https://user-images.githubusercontent.com/31291716/176965297-a6c5175f-42b5-47a2-a053-4afdd7bdb8fb.png">

print function in controlfiltering_shiny.R file mainly print out selected input from shiny sidebar (date, study design, species, strain, sex and etc.) and control animal table. Print commented out inside function, and two print replaced with message function. 
Those option (date, study design, species, strain, sex and etc.) now printing from sendDashboard_shiny.R file with print function used once.
Other file that have print function populateDBFunction.R which print column names and other information. Not sure I should comment that out. 
So basically if user run shiny app calling execSendDashboard function this will print user input (date, study design, species, strain, sex and etc.) in console and dbCreateSchema() function print information about all the table like column name.

**Comment by Yousuf28** [comment]
-  "This might be my hot take but I would [remove the return calls](https://stackoverflow.com/questions/11738823/explicitly-calling-return-in-a-function-or-not) where they aren't being used to set an explicitly stop the evaluation of a function."

4. Main functions of the package written by Bo Larsen and he consistently used return function at the end of each function. 
we can keep or remove the return function at the end of fucntion, either is fine.
 In other place return is used explicitly in condition. 
<img width="1584" alt="Screen Shot 2022-07-01 at 4 44 55 PM" src="https://user-images.githubusercontent.com/31291716/176966684-cc8b12b1-de8d-42db-a5ab-99808137bb62.png">

**Reply by wangw106** [reply]
I removed those Return statements if not used explicitly in condition.

**Comment by Yousuf28** [comment]
- "You're setting a lot of variables to NULL in a few files. You should update this to a single [call to globalVariables](https://github.com/mpadge/spatialcluster/blob/main/R/zzz.R)."

5. NULL replaced with globalVariable as suggested.

<img width="929" alt="Screen Shot 2022-07-04 at 1 48 49 PM" src="https://user-images.githubusercontent.com/31291716/177200699-d9298215-5e9a-4926-a636-c96497ab2993.png">
